source/extensions/filters/network/mysql_proxy/mysql_utils.cc

```diff
@@ -279,6 +273,12 @@ std::vector<uint8_t> OldPassword::signature(const std::string& password,
   return to;
 }
 
+std::vector<uint8_t> NativePassword::generateSeed() {
+  std::vector<uint8_t> res(SEED_LENGTH);
+  RAND_bytes(res.data(), SEED_LENGTH);
+  return res;
```

Comment:
I found that in [cpp 17 standard(class.copy#elision-3)](https://timsong-cpp.github.io/cppwp/n4659/class.copy#elision-3) 
> In the following copy-initialization contexts, a move operation might be used instead of a copy operation:
> - If the expression in a return statement is a (possibly parenthesized) id-expression that names an object with automatic storage duration declared in the body or parameter-declaration-clause of the innermost enclosing function or lambda-expression, or
> - if the operand of a throw-expression is the name of a non-volatile automatic object (other than a function or catch-clause parameter) whose scope does not extend beyond the end of the innermost enclosing try-block (if there is one),
> 
> **overload resolution to select the constructor for the copy is first performed as if the object were designated by an rvalue**

, the lvalue at return statement will not use `copy initialization` if the lvalue type has `move constructor`, use `return std::move(vector)` instead of `return vector` is not helpful, and in the meanwhile use `return std::move(vector)` will prevent `Named Return Value Optimize(NRVO)`.